sys/kern/src/arch/riscv32.rs

```diff
@@ -429,10 +429,9 @@ fn timer_handler() {
 
                 let next = task::select(current, tasks);
                 let next = &mut tasks[next];
-                apply_memory_protection(next);
                 // Safety: next comes from teh task table and we don't use it again
                 // until next kernel entry, so we meet the function requirements.
-                set_current_task(next);
+                crate::task::switch_to(next);
```

Comment:
The naming is odd here, imo. switch_to() reads to me it's actually switching, but that's not what's happening. It's just setting up the bookkeeping for the eventual context switch proper. Maybe that's the semantics of switch_to(), but it seems odd to me.

Reply:
"Performs the architecture-specific bookkeeping to activate `task` on next return to user." which is in the first commit's comments. I guess it's clearly just my opinion. :)

Reply:
Yeah, I hadn't changed the name, but I didn't like it too. Like you mentioned, my assumption on reading the function being used anywhere is that it is switching to the task, but clearly it does not. I thought of `select_next_task`, `setup_next_task`, `configure_next_switch`, `activate_next_task`. Not sure if either of these are any better though. If you have suggestion for a good name we can go with that.